src/pybind/mgr/cephadm/module.py

```diff
@@ -2360,31 +2363,52 @@ def remove_osds(self, osd_ids: List[str],
         """
         Takes a list of OSDs and schedules them for removal.
         The function that takes care of the actual removal is
-        _remove_osds_bg().
+        process_removal_queue().
         """
 
-        daemons = self.cache.get_daemons_by_service('osd')
-        found: Set[OSDRemoval] = set()
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_service('osd')
+        to_remove_daemons = list()
         for daemon in daemons:
-            if daemon.daemon_id not in osd_ids:
-                continue
-            found.add(OSDRemoval(daemon.daemon_id, replace, force,
-                                 daemon.hostname, daemon.name(),
-                                 datetime.datetime.utcnow(), -1))
-
-        not_found = {osd_id for osd_id in osd_ids if osd_id not in [x.osd_id for x in found]}
-        if not_found:
-            raise OrchestratorError('Unable to find OSD: %s' % not_found)
+            if daemon.daemon_id in osd_ids:
+                to_remove_daemons.append(daemon)
+        if not to_remove_daemons:
+            raise NotFoundError('Unable to find OSDs: %s' % osd_ids)
 
-        self.rm_util.queue_osds_for_removal(found)
+        for daemon in to_remove_daemons:
+            try:
+                self.to_remove_osds.enqueue(OSD(osd_id=int(daemon.daemon_id),
```

Comment:
```suggestion
                self.to_remove_osds.enqueue(OSD(daemon=daemon,
```
? 

Why unpack the DD at all?

Reply:
Mh, I did that on purpose avoid a reference to `DaemonDescription` in `OSD`. However, I see the advantage to slim down the func header..

Reply:
hm. As we're just referencing the dds, this shouldn't use any resources. right?

Reply:
Right, my point is not about resources but about entanglement. 

Also it's useful to be able to construct an `OSD` object without a `daemon` ref. See https://github.com/ceph/ceph/blob/5d3979f0c952ca174fc4fae7c78d54a5380895b7/src/pybind/mgr/cephadm/module.py#L2405-L2420